### ISSUE TITLE
Retain option for Lambda@Edge functions?

### ISSUE THREAD

**Issue by iDVB** [serverless-nextjs/serverless-next.js#930]
**Is your feature request related to a problem? Please describe.**
Currently, due to edge function limitations there is no way to delete a sls-next stack without it failing for 2hrs before the replicas are deleted.

**Describe the solution you'd like**
Perhaps providing a "retain" option would allow people to quickly/safely delete the stack minus the functions, and then people could implement their own orphan function cleanup crons. 

**Describe alternatives you've considered**
Until AWS closes this issue, it's likely always going to be one of two solutions:
**A)** let the stack delete fail then setup a cron to delete the stack 2hrs later.
or
**B)** set the edge functions to retail, letting all stack deletes SUCCEED, and then use a cron to clean up those orphan functions 2 hrs later.

**Option B:** seems the most accurate and least disruptive to existing pipelines.

## Other thoughts / Options
The other potential option here for this project is to find a way to "reuse" the same edge functions. (default + images)

**Images Function**
right now it looks like the only unique files in the images is the manifest. If that could be moved out of the function and into an S3 storage to be loaded in and cached then one would only need a single image function for all stacks/sites.

**Default**
This one is more tricky since it contains literally the JS files that are to be used for SSR. It likely is still and option to store these outside of the function and load them in.... but not sure what the performance would be like.

**Comment by kirkness** [comment]
Makes sense, I'd vote option B given this problem will be resolved soon enough (???) by AWS to allow proper stack deletion.

**Comment by ignaciolarranaga** [comment]
Just a side comment, the problem is not just the delete of the stack but the manipulation of the edge functions themselves.

For example I added permissions to the lambda@edge to be able to call another lambdas (specifically from the API) and finally had to move my logic to APIGateway because it was really failing almost all the time due little changes + the mentioned delete problem.

**Comment by davegariepy** [comment]
experiencing this now, option B sounds good
